### DISCUSSION TITLE
Credentials for GCS on Dagster Serverless

### DISCUSSION THREAD

**Discussion by pestrov** [dagster-io/dagster#17559] [Q&A]
**Dagster version**
1.5.5

I am trying to create a Google Cloud Storage resource in Dagster Serverless.

While there is no explicit parameter or documentation for GCS credentials in the dagster-gcp docs, I've tried to reproduce the BigQuery [example](https://docs.dagster.io/integrations/bigquery/reference#providing-credentials-as-configuration).

> For example, if you are using [Dagster Cloud Serverless](https://docs.dagster.io/dagster-cloud/deployment/serverless) you cannot upload a credential file, so must provide your credentials as an environment variable.

I've created a base64 encoded key, as per documentation and have tried multiple options:
`cat ~/.gcp/key.json | base64`

**Option 1**, I've tried to set default env variable _GOOGLE_APPLICATION_CREDENTIALS_ to the base64 encoded key and did not pass any additional params for the GCSResource. 

```python
from dagster_gcp import GCSResource
gcs = GCSResource(project="my-project")
```
When materializing an asset, the gcloud seems to notice the env variable, but seems to interpret it as a filepath:
```
raise exceptions.DefaultCredentialsError
In process 17: google.auth.exceptions.DefaultCredentialsError: File X (base64 content here) was not found. 
```

**Option 2**, just to test the default behavior, I have removed the variable.

```python
gcs = GCSResource(project="my-project")
```

```
In process 17: google.auth.exceptions.DefaultCredentialsError: Your default credentials were not found. To set up Application Default Credentials, see https://cloud.google.com/docs/authentication/external/set-up-adc for more information.
```

**What is the recomeneded way to specify the credentials in Dagster cloud Serverless?**
Is there an issue with my config, or this capability is missing? Definelty seems like I'm missing something but can't find it. 

Could it be that this can be solved if the gcp_credentials parameter from BigQuery resource config be brought to GCS as well? It looks from the [code](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/utils.py) that is a BigQuery specific util at the moment.

**Comment by shalabhc** [comment]
Hi @pestrov - you can try first writing out the credentials and setting `GOOGLE_APPLICATION_CREDENTIALS` to the file that you write out. See https://github.com/dagster-io/dagster/discussions/12183 for more details.

**Reply by pestrov** [reply]
Thanks! I’ve ended up switching to a Hybrid cloud, but this answered you have shared would have defiantly helped!
